nntrainer/src/tensor_dim.cpp

```diff
@@ -57,9 +57,9 @@ void TensorDim::setTensorDim(unsigned int idx, unsigned int value) {
   if (idx > MAXDIM)
     throw std::out_of_range(
       "[TensorDim] Tensor Dimension index should be between 0 and 4");
-  if (value <= 0)
```

Comment:
where do we need the case to remove this.
Current semantics is "allow initialization of TensorDim to 0" but "setting TensorDim to 0 or copying a Tensor/TensorDim with Dim 0 is not allowed"